src/drivers/base.ts

```diff
@@ -0,0 +1,3 @@
+export interface Base {
+  type: string; // MySQl|Postgres|Sqlite
+}
```

Comment:
I prefer the base file to define a unified abstraction of different database clients, similar to:

```ts
export enum ClientType {
  MYSQL = "MYSQL",
  POSTGRESS = "POSTGRESS",
  SQLITE = "SQLITE",
}

export interface DsoConnection {
  query<T = any>(sql: string, params?: any[]): Promise<T>;
  execute<T = any>(sql: string, params?: any[]): Promise<T>;
}

/** Transaction processor */
export interface TransactionProcessor<T> {
  (connection: DsoConnection): Promise<T>;
}

export interface PoolInfo {
  size: number;
  maxSize: number;
  available: number;
}

/**
 * DSO client
 */
export abstract class DsoClient {
  /** get pool info */
  abstract get pool(): PoolInfo;

  /**
   * connect to database
   * @param config config for client
   * @returns Clinet instance
   */
  abstract connect<T>(config: T): Promise<DsoClient>;

  /**
   * excute query sql
   * @param sql query sql string
   * @param params query params
   */
  abstract async query<T = any>(sql: string, params?: any[]): Promise<T>;

  /**
   * excute sql
   * @param sql sql string
   * @param params query params
   */
  abstract async execute<T = any>(sql: string, params?: any[]): Promise<T>;

  /**
   * Execute a transaction process, and the transaction successfully
   * returns the return value of the transaction process
   * @param processor transation processor
   */
  abstract async transaction<T = any>(
    processor: TransactionProcessor<T>
  ): Promise<T>;

  /**
   * close connection
   */
  abstract async close(): Promise<void>;
}

// import { Client } from "../../deps.ts";

// export class MysqlClient extends DsoClient {
//   #client: Client = new Client();

//   async close(): Promise<void> {
//     this.#client.close();
//   }

//   async connect(config: MysqlConfig): Promise<DsoClient> {
//     await this.#client.connect(config);
//     return this;
//   }

//   ...
// }

```

This can avoid using too much template code when dso implements core functions, which is not conducive to maintenance and understanding